File: Data Structures & Algorithms/longest-repeating-substring-with-replacement/submission-0.swift
class Solution {
    func characterReplacement(_ s: String, _ k: Int) -> Int {
        var left = 0
        var map: [Character: Int] = [:]
        let arr = Array(s)
        var length = 0
        var count = 0
        for i in 0..<arr.count {
            map[arr[i], default: 0] += 1
            count = max(map[arr[i]]!, count)
            while((i - left + 1) - count > k) {
                map[arr[left],default: 0] -= 1
                if map[arr[left]] == 0 {
                    map.removeValue(forKey:arr[left])
                }
                left += 1
            }
            length = max(length, i - left + 1)
        }
        return length
    }
}

File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-0.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        left = 0
        seen_char = set()
        length = 0
        for i in range(len(s)):
            while(s[i] in seen_char):
                seen_char.remove(s[left])
                left += 1
            seen_char.add(s[i])
            length = max(i - left + 1,length)
        return length

                



        